mitxgraders/comparers/comparers.py

```diff
@@ -112,30 +121,118 @@ class EqualityComparer(Comparer):
+    reset_default_comparer() methods.
 
     """
     schema_config = Schema({
-        Required('transform', default=None): Nullable(is_callable)
+        Required('transform', default=None): All(
+            Nullable(is_callable),
+            # if f is None, coerce to identity function
+            lambda f: identity_transform if f is None else f
+        )
     })
 
-    def __call__(self, comparer_params_eval, student_eval, utils):
-        expected_input = comparer_params_eval[0]
-
+    @staticmethod
+    def validate(expected_eval, student_eval, utils):
         if hasattr(utils, 'validate_shape'):
             # in numpy, scalars have empty tuples as their shapes
-            shape = tuple() if isinstance(expected_input, Number) else expected_input.shape
+            shape = tuple() if isinstance(expected_eval, Number) else expected_eval.shape
             utils.validate_shape(student_eval, shape)
 
-        # If provided, apply the transform function
-        if self.config['transform'] is not None:
-            expected_input = self.config['transform'](expected_input)
-            student_eval = self.config['transform'](student_eval)
+    def __call__(self, comparer_params_eval, student_eval, utils):
+        expected_eval = comparer_params_eval[0]
+        self.validate(expected_eval, student_eval, utils)
 
-        return utils.within_tolerance(expected_input, student_eval)
+        transform = self.config['transform']
+        expected_eval = transform(expected_eval)
+        student_eval = transform(student_eval)
+
+        return utils.within_tolerance(expected_eval, student_eval)
 
 equality_comparer = EqualityComparer()
 
+class MatrixEntryComparer(CorrelatedComparer):
+    """
+    Default comparer for MatrixGrader. Compares student and instructor matrix
+    evaluations entry-by-entry for equality.
+
+    Configuration
+    =============
+
+        transform (None | function): same as EqualityComparer (default None)
+        entry_partial_credit ('proportional' | number): Determines how partial credit
+            is awarded. If set to 'proportional', then credit is proportional to
+            the number of correct matrix entries. If a numeric value betweem 0 and 1
+            is provided, this flat rate of partial credit is provided as long as
+            some but not all entries are correct. Default is the numeric value 0
+            (no partial credit).
+
+        entry_partial_msg (str): A text string message shown when partial credit
+            is awarded. The string may optionally contain the formatting key {error_indices},
+            which will be replaced with the indices of the incorrect matrix entries.
+            Default value is:
+            "Some matrix entries are incorrect, marked below:\n{error_locations}"
+    """
+
+    default_msg = "Some matrix entries are incorrect, marked below:\n{error_locations}"
+    schema_config = EqualityComparer.schema_config.extend({
```

Comment:
That's a really nice way of building on the previous config without needing to subclass. Good job!